core/commands/object.go

```diff
@@ -348,6 +355,186 @@ Data should be in the format specified by the --inputenc flag.
 	Type: Object{},
 }
 
+var objectNewCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "creates a new object from an ipfs template",
+		ShortDescription: `
+'ipfs object new' is a plumbing command for creating new DAG nodes.
+`,
+		LongDescription: `
+'ipfs object new' is a plumbing command for creating new DAG nodes.
+By default it creates and returns a new empty merkledag node, but
+you may pass an optional template argument to create a preformatted
+node.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("template", false, false, "optional template to use"),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := req.Context().GetNode()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		node := new(dag.Node)
+		if len(req.Arguments()) == 1 {
```

Comment:
i would grab template into a variable first, that way if arg num changes, can change it in only one place. i've been bitten by bugs when doing things this way.

```go
var template string
if len(req.Arguments()) > 0 {
  template = req.Arguments()[0]
}

node := new(dag.Node)
if template != "" {
  ...
}
```

(not a big deal though)